IPython/terminal/embed.py

```diff
@@ -66,14 +65,27 @@ class InteractiveShellEmbed(TerminalInteractiveShell):
     dummy_mode = Bool(False)
     exit_msg = Unicode('')
     embedded = CBool(True)
-    embedded_active = CBool(True)
     should_raise = CBool(False)
     # Like the base class display_banner is not configurable, but here it
     # is True by default.
     display_banner = CBool(True)
     exit_msg = Unicode()
+
+    _inactive_locations = set({})
+    _stack_frames = 3
     
 
+    @property
+    def embedded_active(self):
+        return self._call_location_id not in InteractiveShellEmbed._inactive_locations
+
+    @embedded_active.setter
+    def embedded_active(self, value):
+        if value :
+            InteractiveShellEmbed._inactive_locations.remove(self._call_location_id)
```

Comment:
This should probably have a check if it's currently in the set - `set.remove()` throws KeyError if you try to remove something that's not there.